File: exercism/csharp/etl/ETL.cs
using System.Collections.Generic;


namespace Exercism
{
    public static class ETL
    {
        public static Dictionary<string, int> Transform(Dictionary<int, IList<string>> old)
        {
            var transformed = new Dictionary<string, int>();
            foreach (KeyValuePair<int, IList<string>> e in old) {
                foreach (var piece in e.Value) {
                    transformed.Add(piece.ToLower(), e.Key);
                }
            }
            return transformed;
        }
    }
}

File: exercism/cpp/etl/etl.cpp
#include "etl.h"

std::map<char, int> etl::transform(std::map<int, std::vector<char>> old) {
  std::map<char, int> transformed;
  for (auto const &kv : old) {
    int n = kv.first;
    for (const char &ch : kv.second) {
      transformed[tolower(ch)] = n;
    }
  }
  return transformed;
}
